descriptors/descriptors.go

```diff
@@ -35,6 +41,33 @@ func (d *Descriptor) String() string {
 	return d.mod.descriptorString(d.ptr)
 }
 
-func (d *Descriptor) AddressAt(network Network, multipathIndex uint32, derivationIndex uint32) (string, error) {
-	return d.mod.descriptorAddressAt(d.ptr, network, multipathIndex, derivationIndex)
+func (d *Descriptor) AddressAt(network Network, multipathIndex uint32,
+	derivationIndex uint32) (string, error) {
+
+	return d.mod.descriptorAddressAt(
+		d.ptr, network, multipathIndex, derivationIndex,
+	)
+}
+
+type Miniscript struct {
+	ParsedTypes string
+	OpCodes     uint64
+}
+
+func ParseMiniscript(script string) (*Miniscript, error) {
+	mod := getWasmMod()
+	jsonRes, err := mod.miniscriptParse(script)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Miniscript{
+		ParsedTypes: jsonRes.Types,
+		OpCodes:     jsonRes.OpCodes,
+	}, nil
+}
+
+func CompileMiniscript(script string) ([]byte, error) {
```

Comment:
"Compile" is a bit overloaded as a name (and reminds of the policy compiler). MiniscriptToPkScript or something like that would be good. But again, why do we need this when we have NewDescriptor() which can serve the same purpose?

Reply:
My goal was to replace https://github.com/btcsuite/btcd/pull/1987 with this PR. And for building a wallet with descriptor capability (or re-write `lnd` to use miniscript expressions instead of hard-coded BTC scripts for on-chain operations), I thought we'd need to be able to just turn an expression into a `pkScript`. But you're right, maybe it is enough to just add a `ToPkScript()` method to the descriptor. I'll look into that.

Reply:
In the meantime, you could also just move this function to `descriptors_test.go` (and unexport it), in case you find it valuable to keep the compile tests.